homeassistant/components/aurora_abb_powerone/__init__.py

```diff
@@ -29,9 +31,46 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     comport = entry.data[CONF_PORT]
     address = entry.data[CONF_ADDRESS]
     serclient = AuroraSerialClient(address, comport, parity="N", timeout=1)
+    # To handle yaml import attempts in darkeness, (re)try connecting only if
+    # unique_id not yet assigned.
+    if entry.unique_id is None:
+        try:
+            res = await hass.async_add_executor_job(
+                validate_and_connect, hass, entry.data
+            )
+        except AuroraError as error:
+            if "No response after" in str(error):
+                raise ConfigEntryNotReady("No response (could be dark)") from error
+            raise ConfigEntryNotReady(
+                "Failed to connect to inverter: %s" % error
+            ) from error
+        except OSError as error:
+            if error.errno == 19:  # No such device.
+                raise ConfigEntryNotReady(
+                    "Failed to connect to inverter: no such COM port"
+                ) from error
```

Comment:
If there is no such device, it can't recover and so we should mark setup as failed instead of "failed but might work in a bit"

```suggestion
                return False
```

Reply:
If device is a USB-RS485 adaptor perhaps it will work when it is plugged back in?

But retry every 5sec is probably not a good idea in this case.

I don't really mind. My initial approach was 'no response' is the only one we want to allow deferring for.  All others are errors which should stop.  But that means restart HA is the only way to 'try again'.

Reply:
I have switched them to log errors then return `False`. Hopefully that satisfies all.